Library/Homebrew/software_spec.rb

```diff
@@ -107,6 +108,10 @@ def resource(name, klass = Resource, &block)
     end
   end
 
+  def get_resource
+    @resource
+  end
```

Comment:
Yes, there is a reason for that. `SoftwareSpec#resourse` already exists.
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/software_spec.rb#L99-L108